packages/react-components/react-breadcrumb/docs/Spec.md

```diff
@@ -2,54 +2,479 @@
+| Breadcrumb        | Wrapper for the Breadcrumb component. Contains `nav` and `ol` elements. |
+| BreadcrumbDivider | Divider component                                                       |
+| BreadcrumbItem    | Contains `li` element as a wrapper. Can contain button and link         |
+| BreadcrumbMenu    | Can contain subfolders or hidden items.                                 |
+| BreadcrumbLink    | Can be used as Button or Link                                           |
+
+### Breadcrumb
+
+#### Anatomy
+
+![visual anatomy of the Breadcrumb component](./assets/breadcrumb-anatomy.png)
+
+#### DOM
+
+```HTML
+<nav aria-label="breadcrumb">
+  <ol>
+  {children}
+  </ol>
+</nav>
+```
+
+#### Breadcrumb structure:
+
+- root `nav` element
+- list `ol` element
+- `divider` - a BreadcrumbDivider component
+- `menu` - a BreadcrumbMenu component
+
+#### API
+
+| Property          | Values                                  | Default       | Purpose                                                                                  |
+| ----------------- | --------------------------------------- | ------------- | ---------------------------------------------------------------------------------------- |
+| appearance        | `transparent`, `subtle`                 | `transparent` | Sets appearance                                                                          |
+| menu              | _slot_                                  | undefined     | Sets custom dropdown menu                                                                |
+| focusMode         | `off`, `no-tab`, `tab-exit`, `tab-only` | `off`         | Sets the focus behavior for the Breadcrumb.                                              |
+| maxDisplayedItems | number                                  | undefined     | If items count more than `maxDisplayedItems` only this number of items will be displayed |
+| overflowIndex     | number                                  | 1             | Index of item where BreadcrumbMenu should be shown                                       |
+| size              | `small`, `medium`, `large`              | `medium`      | Defines size of the Breadcrumb                                                           |
+
+### BreadcrumbItem
+
+#### Anatomy
+
+![visual anatomy of the BreadcrumbItem component](./assets/breadcrumb-item-anatomy.png)
+
+BreadcrumbItem can be:
+
+- Button
+- Link
+- Non-clickable content
+- Dropdown Menu
+- Icon (optional)
+
+It can contain a tooltip.
+
+#### DOM
+
+Non-clickable element
+
+```HTML
+<li>
+  {children}
+</li>
+```
+
+Link
+
+```HTML
+<li>
+  <a href="#">
+    {children}
+  </a>
+</li>
+```
+
+Button
+
+```HTML
+<li>
+  <button>
+    {children}
+  </button>
+</li>
+```
+
+Usage
+
+```jsx
+<BreadcrumbItem>
+  Item 1
+</BreadcrumbItem>
+<BreadcrumbItem onClick={() => console.log('smth...')}>
+  Item 2
+</BreadcrumbItem>
+<BreadcrumbItem href="#">
+  Item 3
+</BreadcrumbItem>
+```
```

Comment:
I honestly would prefer N*-like implementation:

```tsx
<BreadcrumbItem>
  <BreadcrumbLink href="">Home</BreadcrumbLink>
</BreadcrumbItem>
<BreadcrumbItem>
  <BreadcrumbButton>Home</BreadcrumbButton>
</BreadcrumbItem>
```

- It keeps the surface of `BreadcrumbItem` clearer
- It avoids slots problem, as otherwise you will have then only two options:
  - `Input`-like behavior with root slot when some props go `button`/`a` and some remain on `li`
  - Default with slots, but then you will need to have either two slots (`<BreadcrumbItem button={} />` & `<BreadcrumbItem anchor={} />`) or a single one that implements both (`<BreadcrumbItem action={} />`)

Reply:
I agree. I added Button and Link as separate components